apps/publisher/themes/default/js/taxonomy-browser.js

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+const BASE_URL = caramel.context + '/apis/taxonomies';
+const TAXONOMY_SELECT = '.taxonomy-select';
+const COLUMN_SELECTOR = '.taxonomy-select-window';
+const BREADCRUMB_SELECTOR = '.taxonomy-breadcrumb > ol';
+const TAXONOMY_DROPDOWN = '#taxonomy-dropdown-menu';
+const TAXONOMY_BROWSER = '.taxonomy-browser';
+const SELECTED_CONTAINER = '.selected-taxonomy-container';
+const SELECTED_CONTENT = '.selected-taxonomy-content';
+const BACK_LINK = '<li class="back" style="display: none;"><a href="#">...</a></li>';
+const RIGHT_ARROW = ' <i class="icon fw fw-right-arrow"></i>';
+
+var windowObject = {};
+var selectedTaxonomy = [];
+var displayValue = [];
+var columnsCount;
+
+/**
+ * Initializes the taxonomy browser window column size based on the window size.
+ */
+var initWindowColumns = function () {
+    if ($(window).width() < 768) {
+        columnsCount = 1;
+    }
+    else if ($(window).width() >= 768 && $(window).width() <= 992) {
+        columnsCount = 2;
+    }
+    else if ($(window).width() > 992 && $(window).width() <= 1200) {
+        columnsCount = 3;
+    }
+    else {
+        columnsCount = 4;
+    }
+};
```

Comment:
@thushara35 this variable represents the number of columns in the taxonomy browser. IMHO if we use a more lengthy name that would affect the clarity of the code. Therefore instead of renaming the variable, I have explained the use of the variable at the point of initialization in order to increase the maintainability of the code. #841 (9918ae8). WDYT of this approach?